near/omni-types/src/prover_result.rs

```diff
@@ -3,17 +3,24 @@ use near_sdk::json_types::U128;
 use near_sdk::serde::{Deserialize, Serialize};
 use near_sdk::AccountId;
 
-use crate::{OmniAddress, TransferMessage};
+use crate::{Fee, OmniAddress, TransferId};
 
 #[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Debug, Clone)]
 pub struct InitTransferMessage {
-    pub transfer: TransferMessage,
+    pub origin_nonce: U128,
+    pub token: OmniAddress,
+    pub amount: U128,
+    pub recipient: OmniAddress,
+    pub fee: Fee,
+    pub sender: OmniAddress,
+    pub msg: String,
     pub emitter_address: OmniAddress,
 }
 
 #[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Debug, Clone)]
 pub struct FinTransferMessage {
-    pub nonce: U128,
+    pub payload_nonce: U128,
```

Comment:
I think maybe we should use this nonce purely to prevent replays on destination chains and exclude it from everywhere else. TransferId should be sufficient for every other use case and having two identifiers just adds confusion